web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -288,11 +328,39 @@ export function Navigation() {
     [debouncedSection]
   );
 
-  const resetExpandedSection = useCallback((closeAfterDelay = true) => {
-    setIsClosing(closeAfterDelay);
-    setPreviousExpandedSection(null);
-    setTargetSection(null);
-  }, []);
+  const combinedSideNavSections = useMemo(
+    () => [resourcesSection, ...navSections],
+    [resourcesSection, navSections]
+  );
+  const currentPageSection = useMemo(() => {
+    return combinedSideNavSections.find(
+      section => section.category === currentView?.category
+    );
+  }, [combinedSideNavSections, currentView]);
+
+  const resetExpandedSection = useCallback(
+    (closeAfterDelay = true) => {
+      if (stickyMode && currentPageSection) {
+        setPreviousExpandedSection(debouncedSection);
+        setTargetSection(currentPageSection);
+      } else {
+        setIsClosing(closeAfterDelay);
+        setPreviousExpandedSection(null);
+        setTargetSection(null);
+      }
+    },
+    [currentPageSection, stickyMode, debouncedSection]
+  );
+
+  useEffect(() => {
+    if (stickyMode) {
+      if (currentPageSection) {
+        handleSetExpandedSection(currentPageSection);
+      } else {
+        resetExpandedSection();
+      }
+    }
```

Comment:
The number of `ifs` make it quite hard to read. We have almost the same conditions in both the `useEffect` and `resetExpandedSection`, expanding all possible branches in mind requires some effort. Do you think we could simplify it somehow? I wanted to propose something but I feel don't fully understand the logic.